fix the numNodes method

sjt003 opened the pull request.

letsgetrandy committed:
fix the numNodes method

sjt003 reviewed (commented):
I approve

mikenichols commented on lib/graph.js:
```diff
@@ -44,7 +44,7 @@ DirectedGraph.prototype.getNode = function (id) {
 DirectedGraph.prototype.numNodes = function () {
     'use strict';
-    return nodes.length;
+    return this.nodes.length;
```
Is there a test for this/can we write one?

sjt003 committed:
fix the undirected graph numNodes method

missing 'this' (same as the numNodes method of DirectedGraph)

sjt003 committed:
add tests for `numNodes` fixes

test functionality of `numNodes` methods for both `UndirectedGraph` and
`DirectedGraph`.
- [x] null graph tests
- [x] non-null graph

mikenichols reviewed (approved):
LGTM 👍

sjt003 committed:
add circle.yml with asdf node v @ 7.5.0

right now this will be running the mocha tests

sjt003 committed:
add circleci asdf install script from imp_backend

sjt003 committed:
add `numNodes` method documentation

I fixed the method; therefore, I should document it if it is not already documented.
